apps/core/components/header/cart-icon.tsx

```diff
@@ -0,0 +1,44 @@
+'use client';
+
+import { Badge } from '@bigcommerce/components/badge';
+import { ShoppingCart } from 'lucide-react';
+import { useEffect, useState } from 'react';
+import { z } from 'zod';
+
+export const CartQuantityResponseSchema = z.object({
+  count: z.number(),
+});
+
+interface CartIconProps {
+  count?: number;
+}
+
+export const CartIcon = ({ count }: CartIconProps) => {
+  const [fetchedCount, setFetchedCount] = useState<number | null>();
+  const computedCount = count ?? fetchedCount;
+
+  useEffect(() => {
+    async function fetchCartQuantity() {
+      const response = await fetch(`/api/cart-quantity/`);
+      const parsedData = CartQuantityResponseSchema.parse(await response.json());
+
+      setFetchedCount(parsedData.count);
+    }
+
+    if (count === undefined) {
+      void fetchCartQuantity();
+    }
+  }, [count]);
+
+  if (!computedCount) {
+    return <ShoppingCart aria-label="cart" />;
```

Comment:
```suggestion
    return <ShoppingCart aria-label="cart" />
```

Reply:
Prettier is actually adding this one here (but fixed the other one)